compiler/rustc_middle/src/ty/typeck_results.rs

```diff
@@ -594,11 +594,24 @@ pub struct CanonicalUserTypeAnnotation<'tcx> {
 /// Canonical user type annotation.
 pub type CanonicalUserType<'tcx> = Canonical<'tcx, UserType<'tcx>>;
 
-impl<'tcx> CanonicalUserType<'tcx> {
```

Comment:
I would prefer using an extension trait I think, `UserType::is_identity` seems wrong and easy to misuse